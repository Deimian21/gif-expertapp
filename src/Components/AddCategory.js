import React, { useState } from 'react'
import Proptypes from 'prop-types';

export const AddCategory = ({setCategory}) => {
    const [imputValiu, setimputValiu] = useState('');
    const handleInputChange= (e) =>{
     setimputValiu(e.target.value);
    }
    const handleSumbit = (e) =>{
      e.preventDefault();
      if (imputValiu.trimEnd().length>2) {
        setCategory( cats => [imputValiu,...cats]);
        setimputValiu('');
      }
      
    }
    return (
        <form onSubmit={handleSumbit}>
            
            <h2> AÃ±adir componete</h2>
            <input type='text' value={imputValiu} onChange={handleInputChange}></input>
            </form>
    )
}

AddCategory.propTypes ={
  setCategory:Proptypes.func.isRequired
}
