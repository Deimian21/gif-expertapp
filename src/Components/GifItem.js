import React from 'react'
import PropTypes from 'prop-types';
export const GifItem = ({url,title}) => {
    //console.log(props);
    return (
        <div className='animate__fadeIn'>
           <img src={url} alt={title}/>
           <p>{title}</p>
        </div>
    )
}
GifItem.propTypes= {
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
    }