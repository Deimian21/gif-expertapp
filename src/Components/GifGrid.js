//import { logDOM } from '@testing-library/react';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
     const {data:images,loading} = useFetchGifs(category);

    return (
        <>
        <h3 className='animate__fadeIn'>{category}</h3>
        { loading && <p>Loading</p>}
        <div className='card-grid'>
          
            { 

            images.map( img =>(
            <GifItem 
            key={img.id}
            {...img}/>
            ) )

            }
        </div>
        </>
    )
}
GifGrid.propTypes = {
    category:PropTypes.string.isRequired
}
