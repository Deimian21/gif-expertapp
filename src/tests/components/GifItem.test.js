import React from 'react';
import { shallow } from 'enzyme';
import { GifItem } from '../../Components/GifItem';

describe('Pruebas', () => {
    const title='Prueba';
    const url='https://algo.jpg';
    const wrapper = shallow( <GifItem title={title} url={url} /> ); 
    test('debe devolver el componente ', () => {

        
        expect(wrapper).toMatchSnapshot();
   
});
test('debe de tener un parrafo con el title ', () => {
    const p =wrapper.find('p');
    expect(p.text().trim()).toBe(title);
})
test('debe tener la imagen igual al url y alt de los prots', () => {
    const img=wrapper.find('img');
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)  
})
 test('debe tener la clase faid int ', () => {
     const div= wrapper.find('div');
     expect(div.prop('className')).toBe('animate__fadeIn');

 })
 


})
 