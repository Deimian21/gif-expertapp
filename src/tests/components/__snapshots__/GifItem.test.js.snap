// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas debe devolver el componente  1`] = `
<div
  className="animate__fadeIn"
>
  <img
    alt="Prueba"
    src="https://algo.jpg"
  />
  <p>
    Prueba
  </p>
</div>
`;
